### PR TITLE
#110 fix context comments

### PR DESCRIPTION
**Pull request by abynenkov-ib**

### CODE REVIEW

File: pkg/compiler/rego/rego.go [hunk 1]
```diff
@@ -170,13 +181,20 @@ func (c *CompilerRego) linearizeContext(stmt *ast.ContextStatement) []*ast.Actio
...
-				for _, icond := range stmt.Conditions {
-					ctx.Conditions = append(ctx.Conditions, icond)
+
+				for _, icond := range stmt.Conditions { // add conditions defined for 'parent' context
+					// but only in case it's not blank, mean parent does not looks like context {}...
+					if icond.Subject != nil || icond.Where != nil {
+						ctx.Conditions = append(ctx.Conditions, icond)
+					}
 				}
+
```
Comment (abynenkov-ib) [inline pkg/compiler/rego/rego.go:196]: Suggestion (also parser.validateContextStatement should be adjusted):
```
				//if Verb is not defined for child context - set it from parent
				if ctx.Verb == nil {
					ctx.Verb = stmt.Verb
				}
				// and the same for TypePattern
				if ctx.TypePattern == nil {
					ctx.TypePattern = stmt.TypePattern
				}

```

**Review by rchowinfoblox — APPROVED** [review]
LGTM

**Review by wplu — CHANGES_REQUESTED** [review]
a lot of these nil checks need to use IsNilInterface()

File: pkg/compiler/rego/rego.go (hunk 1 above)
Comment (wplu) [reply pkg/compiler/rego/rego.go:196]: let's do that in next pr

**Review by wplu — APPROVED** [review]

**Pull request merged**